torch/lib/THD/base/data_channels/DataChannelGloo.hpp

```diff
@@ -103,7 +103,16 @@ struct DataChannelGloo : DataChannel {
   std::string _addr;
   port_type _port;
   rank_type _num_processes; // Number of processes in network
-  std::shared_ptr<::gloo::transport::Device> _device;
+  /**
+   * The list of network devices (such as Infiniband) that will be used by Gloo.
+   * Currently Gloo only supports a single network device. Therefore:
+   *
+   * _deviceList.size() will always be equal or less than 1.
+   *
+   * We make it a vector for the purpose of future extension to support multiple
+   * network devices.
+   */
+  std::vector<std::shared_ptr<::gloo::transport::Device>> _deviceList;
```

Comment:
Uh do we have to change this now if we're not implementing it? It seems to unnecessarily complicate things. It can't even be zero - we should throw an error if we don't have a device!

Reply:
I think it should be fine to leave it this way since there can be multiple devices, and we can easily implement it later.

Yeah, the error is thrown here in DataChannelGloo.cpp:82